src/webui/webapplication.cpp

```diff
@@ -373,6 +373,9 @@ void WebApplication::doProcessRequest()
                 }
             }
             break;
+        case QMetaType::UnknownType:
+            status(204);
```

Comment:
shouldn't you add `Q_ASSERT(result.data.isNull());`?
`QMetaType::UnknownType` may indicate the contained type is unregistered instead of it being empty.

Reply:
Won't this result in qBittorrent exiting if it's hit? Based on the `QMetaType` [enum](https://doc.qt.io/qt-6/qmetatype.html#Type-enum), it's not clear to me what data would have to be returned for the type to still be unknown.

Reply:
>Won't this result in qBittorrent exiting if it's hit? 

It won't: https://doc.qt.io/qt-6/qtassert-qtcore-proxy.html#Q_ASSERT

>Based on the QMetaType [enum](https://doc.qt.io/qt-6/qmetatype.html#Type-enum), it's not clear to me what data would have to be returned for the type to still be unknown.

The type isn't the point, the point is it should not contain any data, i.e. it should be null (from QVariant point of view).

Reply:
> The type isn't the point, the point is it should not contain any data, i.e. it should be null (from QVariant point of view).

It is empty.
>In current architecture if "action" returns no data (i.e. it doesn't call `setResult()`) the data is of type `QMetaType::UnknownType` at this point

Reply:
Of course we could check it for `isNull()` instead of `QMetaType::UnknownType`:
```c++
if (data.isNull())
    status(204);
else
    // check type
```

Reply:
> > Won't this result in qBittorrent exiting if it's hit?
> 
> It won't: https://doc.qt.io/qt-6/qtassert-qtcore-proxy.html#Q_ASSERT

But `Q_ASSERT` calls `qFatal`, which will core dump. Or are we overriding the default message handler?



> Of course we could check it for `isNull()` instead of `QMetaType::UnknownType`:
> 
> ```c++
> if (data.isNull())
>     status(204);
> else
>     // check type
> ```

This seems like the best option. Will update.

Reply:
>But Q_ASSERT calls qFatal, which will core dump. Or are we overriding the default message handler?

You probably missed this:
>It does nothing if QT_NO_DEBUG was defined during compilation.

qbt do define it.

>In current architecture if "action" returns no data (i.e. it doesn't call setResult()) the data is of type QMetaType::UnknownType at this point
>
>In other words the result QVariant is uninitialized (i.e. it doesn't contain any data).

It doesn't rule out human error which called `setResult()` with unregistered datatype.